### PR TITLE
UI: Use a compatibility questionnaire

### PR DESCRIPTION
**Pull request by german77**
Use a questionnaire and determine the compatibility with the answers. Instead of just selecting from a radio button list.

This adds the option to have more detailed compatibility reports.

### CODE REVIEW

File: src/yuzu/compatdb.h [hunk 1]
```diff
@@ -12,19 +12,30 @@ namespace Ui {
...
     ~CompatDB();
+    int nextId() const override;
 
 private:
     QFutureWatcher<bool> testcase_watcher;
 
     std::unique_ptr<Ui::CompatDB> ui;
 
     void Submit();
+    CompatibilityStatus CalculateCompatibility();
```
Comment (lioncash) [inline src/yuzu/compatdb.h:38]: This can be a const member function

File: src/yuzu/compatdb.ui [hunk 2]
```diff
@@ -201,12 +130,619 @@
...
+         <widget class="QCheckBox" name="checkbox_Freeze_softlock">
+           <property name="text">
+             <string>The game softlocks</string>
+           </property>
+         </widget>
+       </item>
+       <item row="7" column="0">
+         <widget class="QCheckBox" name="checkbox_Freeze_memoryleak">
+           <property name="text">
+             <string>The has memory leaks</string>
```
Comment (v1993) [inline src/yuzu/compatdb.ui:323]:
```suggestion
             <string>The game has memory leaks</string>
```

File: src/yuzu/compatdb.ui [hunk 3]
```diff
@@ -201,12 +130,619 @@
...
+         <widget class="QCheckBox" name="checkbox_Graphical_corrupted">
+           <property name="text">
+             <string>Corruped textures</string>
+           </property>
+         </widget>
+       </item>
+       <item row="4" column="0">
+         <widget class="QCheckBox" name="checkbox_Graphical_fliquer">
+           <property name="text">
+             <string>Fliquering graphics</string>
```
Comment (v1993) [inline src/yuzu/compatdb.ui:454]: I'm no English pro, but shouldn't this be "flickering"?

File: src/yuzu/compatdb.ui [hunk 4]
```diff
@@ -201,12 +130,619 @@
...
+         <widget class="QCheckBox" name="checkbox_Audio_faster">
+           <property name="text">
+             <string>Audio is faster</string>
+           </property>
+         </widget>
+       </item>
+       <item row="3" column="0">
+         <widget class="QCheckBox" name="checkbox_Audio_slower">
+           <property name="text">
+             <string>Audio is slower</string>
```
Comment (v1993) [inline src/yuzu/compatdb.ui:606]: "Audio is too fast/too slow"

File: src/yuzu/compatdb.ui [hunk 5]
```diff
@@ -201,12 +130,619 @@
...
+         <widget class="QCheckBox" name="checkbox_Graphical_missing">
+           <property name="text">
+             <string>Missing textures</string>
+           </property>
+         </widget>
+       </item>
+       <item row="3" column="0">
+         <widget class="QCheckBox" name="checkbox_Graphical_corrupted">
+           <property name="text">
+             <string>Corruped textures</string>
```
Comment (IanHollow) [inline src/yuzu/compatdb.ui:447]:
```suggestion
             <string>Corrupted textures</string>
```

**Pull request closed without merging**